lib/BackgroundJob/ZipJob.php

```diff
@@ -29,22 +29,40 @@
 use OCA\FilesZip\Service\ZipService;
 use OCP\AppFramework\Utility\ITimeFactory;
 use OCP\BackgroundJob\QueuedJob;
+use Psr\Log\LoggerInterface;
 
 class ZipJob extends QueuedJob {
 
 	/** @var ZipService */
 	private $zipService;
+	/** @var LoggerInterface */
+	private $logger;
 
-	public function __construct(ITimeFactory $timeFactory, ZipService $zipService) {
+	public function __construct(ITimeFactory $timeFactory, ZipService $zipService, LoggerInterface $logger) {
 		parent::__construct($timeFactory);
 		$this->zipService = $zipService;
+		$this->logger = $logger;
 	}
 
-	protected function run($argument) {
-		$uid = $argument['uid'];
-		$fileIds = $argument['fileIds'];
-		$target = $argument['target'];
+	public function getUid(): string {
+		return $this->argument['uid'];
+	}
+
+	public function getFileIds(): array {
+		return $this->argument['fileIds'];
+	}
 
-		$this->zipService->zip($uid, $fileIds, $target);
+	public function getTarget(): string {
+		return $this->argument['target'];
+	}
+
+	protected function run($argument) {
+		try {
+			$file = $this->zipService->zip($this->getUid(), $this->getFileIds(), $this->getTarget());
+			$this->zipService->sendNotificationOnSuccess($this, $file);
```

Comment:
If the notification dispatch is called from here, I would have put the method definition in that class. Or I would make the calls inside the zip service like for `buildScheduledNotification`.

Reply:
I'd still prefer it here since the notification is more related to the job than the actual zipping process, in case we need to reuse the zip creating anywhere outside e.g. when we decide to perform quick compressions of small files immediately where we would not need a notification.

With that I wanted to keep the notification creation logic in a shared place since the buildScheduledNotification would be reused by the success/failure ones to cleanup the pending notification.

Though for a bit more cleaner structure let me move the notification handling to a separate service.